### PR TITLE
haxe: fix build for Linux

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/Homebrew/homebrew-core/runs/3386974064?check_suite_focus=true
```
# Run eval $(opam env --switch=default) to update the current shell environment
==> opam config exec -- opam pin add haxe /tmp/haxe-20210821-15715-2alcvn --no-action
Warning:  exec was deprecated in version 2.1 of the opam CLI. Use opam exec instead or set OPAMCLI environment variable to 2.0.
Fatal error:
"rsync": command not found.
```

### CODE REVIEW

File: Formula/haxe.rb [hunk 1]
```diff
@@ -39,8 +39,13 @@ class Haxe < Formula
 
   uses_from_macos "m4" => :build
   uses_from_macos "perl" => :build
+  uses_from_macos "rsync" => :build
   uses_from_macos "unzip" => :build
 
+  on_linux do
+    depends_on "node" => :test
```
Comment (cho-m) [inline Formula/haxe.rb:46]: Alternatively, can disable JavaScript test or change to another language. On macOS, `osascript` is pre-installed and can run JavaScript.

Comment (danielnachun) [reply Formula/haxe.rb:46]: Since this is only a test dependency, I think it's fine.

**Review by danielnachun — DISMISSED** [review]

File: Formula/haxe.rb [hunk 2]
```diff
@@ -109,7 +111,13 @@ class HelloWorld {
       }
     EOS
     system "#{bin}/haxe", "-js", "out.js", "-main", "HelloWorld"
-    _, stderr, = Open3.capture3("osascript -so -lJavaScript out.js")
-    assert_match "Hello world!", stderr.strip
+    output = nil
+    on_macos do
+      _, output, = Open3.capture3("osascript -so -lJavaScript out.js")
```
Comment (dawidd6) [inline Formula/haxe.rb:116]: This can be `shell_output` too I think

Comment (cho-m) [reply Formula/haxe.rb:116]: Should be possible if we redirect stderr (`2>&1`). Not sure why it outputs to stderr though. I tried changing `-s` value but didn't do anything.

**Review by dawidd6 — APPROVED** [review]

**Pull request merged**